### ISSUE TITLE
[BUG] Inconsistency in online_shapelet_distance

### ISSUE THREAD

**Issue by margauxgeuzaine** [sktime/sktime#5480]
There is an inconsistency in sktime/transformations/panel/_shapelet_transform_numba.py, between line 24 where "std = (sum2 - mean * mean * length) / length" and line 56 where "std = math.sqrt((sums2[n] - mean * mean * length) / length)". 

Because of that, some time series are treated differently than others regarding normalization. The former are actually divided by the variance, and the latter are divided by the standard deviation. The second way of doing is supposed to be the right one.

Version 0.24.0 and before.

**Comment by fkiraly** [comment]
Yes, looks like a logic bug. Second way looks correct - fix seems straightforward, would appreciate a PR.
